test/geometry/epipolar/test_projection.py

```diff
@@ -138,3 +138,54 @@ def test_gradcheck(self, device):
         F_mat = torch.rand(1, 3, 3, device=device, dtype=torch.float64, requires_grad=True)
         assert gradcheck(epi.projections_from_fundamental,
                          (F_mat,), raise_exception=True)
+
+
+class TestKRtFromProjection:
+    def test_smoke(self, device, dtype):
+        P = torch.randn(1, 3, 4, device=device, dtype=dtype)
+        K, R, t = epi.KRt_from_projection(P)
+        assert K.shape() == (1, 3, 3)
+        assert R.shape() == (1, 3, 3)
+        assert t.shape() == (1, 3, 1)
+
+    @pytest.mark.parametrize("batch_size", [1, 2, 4])
+    def test_shape(self, batch_size, device, dtype):
+        B: int = batch_size
+        P = torch.rand(B, 3, 3, device=device, dtype=dtype)
+        K, R, t = epi.KRt_from_projection(P)
+        assert K.shape() == (B, 3, 3)
+        assert R.shape() == (B, 3, 3)
+        assert t.shape() == (B, 3, 1)
+
+    def test_simple(self, device, dtype):
+        P = torch.Tensor([[
+            [232., 71., 267., 352.]
```

Comment:
from were did you get this numbers ?

Reply:
These were the numbers generated randomly and fed into  OpenCV's function [decomposeProjectionMatrix](https://github.com/opencv/opencv/blob/60354e39018ac82d230d0d021cdc77343fc84ee8/modules/calib3d/src/calibration.cpp#L3271).
The `K`, `R` and `t` returned from this function are used to verify with my implementation.